lib/utils.rb

```diff
@@ -56,3 +56,33 @@ def exit_unless_dir(path)
   end
   return true
 end
+
+def expand_node_ranges(nodes)
+  new_nodes = []
+  nodes.each do |node|
+    if node.match(/.*\[[0-9]+.*[0-9]+\]$/)
+      m = node.match(/^(.*)\[(.*)\]$/)
+      prefix = m[1]
+      suffix = m[2]
+      ranges = suffix.split(',')
+      ranges.each do |range|
+        if range.match(/-/)
+          num_1, num_2 = range.split('-')
+          padding = num_1.match(/^0+/)
+          unless num_1 <= num_2
+            $stderr.puts "Invalid node range #{range}"
```

Comment:
While I'm here... It might be better to raise an exception rather than a `puts`/`exit`.

Reply:
Ah, thanks for the advice, in that case I should make the change I make in quite a few places 

\> greps for 'exit' and sweats nervously